src/frontends/sql/decoder_context_test.cc

```diff
@@ -166,4 +167,67 @@ INSTANTIATE_TEST_SUITE_P(DecodeMergeOpTest, DecodeMergeOpTest,
                          Combine(ValuesIn(kSampleInputVectors),
                                  ValuesIn(kSampleInputVectors)));
 
+class DecodeTagTransformTest
+    : public TestWithParam<
+          std::tuple<ir::Value, absl::string_view, std::vector<ir::Value>,
+                     std::vector<uint64_t>>> {
+ protected:
+  DecodeTagTransformTest() : ir_context_(), decoder_context_(ir_context_),
+  value_vec_(std::get<2>(GetParam())) {
+    // The id_sequence_vec is an id sequence that may have more IDs than we
+    // need for giving each value an ID. We will put a value_vec.size()
+    // prefix of it into input_id_vec_ so input_id_vec_ contains only those IDs
+    // associated with a value.
+    const std::vector<uint64_t> &id_sequence_vec = std::get<3>(GetParam());
+
+    input_id_vec_.reserve(value_vec_.size());
```

Comment:
`input_id_vec_` seems to contain the same `id_sequence_vec_`? Why do we need to reconstruct it?

Reply:
`id_sequence_vec` is, most likely, too long. I have a few ID sequences that are long enough for any of my example lists of values; the idea is that values are associated with IDs in the order described by the ID sequence. So only a prefix of this sequence of IDs will actually become precondition IDs.